src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -582,6 +656,11 @@ public override ValueTask<int> ReadAsync(Memory<byte> destination, CancellationT
                         return new ValueTask<int>(Task.FromException<int>(new ObjectDisposedException(nameof(Http2ReadStream))));
                     }
 
+                    if (http2Stream._abortException != null)
+                    {
+                        return new ValueTask<int>(Task.FromException<int>(new HttpRequestException(SR.net_http_client_execution_error, http2Stream._abortException)));
```

Comment:
I think this should be IOException instead of HttpRequestException.  We generally throw IOExceptions out of the read/write operations on these streams, e.g. https://github.com/dotnet/corefx/blob/a28444b71a45171c07eb28a5d82086a4589665fe/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ContentLengthReadStream.cs#L42

Reply:
updated.